src/semgrep_agent/targets.py

```diff
@@ -124,22 +126,32 @@ def get_git_status(self) -> GitStatus:
             if code == StatusCode.Untracked or code == StatusCode.Ignored:
                 continue
 
-            # The following detection for unmerged codes comes from `man git-status`
-            if code == StatusCode.Unmerged:
-                unmerged.append(self._fname_to_path(repo, fname))
-            if (
-                code[0] == StatusCode.Renamed
-            ):  # code is RXXX, where XXX is percent similarity
-                removed.append(self._fname_to_path(repo, fname))
-                fname = status_output[2]
-                trim_size += 1
-                added.append(self._fname_to_path(repo, fname))
-            if code == StatusCode.Added:
-                added.append(self._fname_to_path(repo, fname))
-            if code == StatusCode.Modified:
-                modified.append(self._fname_to_path(repo, fname))
-            if code == StatusCode.Deleted:
-                removed.append(self._fname_to_path(repo, fname))
+            resolved_name = self._fname_to_path(repo, fname)
+
+            # If file is symlink to directory, skip
+            absolute_name = Path(repo.working_tree_dir) / fname
+            if absolute_name.is_symlink() and resolved_name.is_dir():
+                click.echo(
+                    f"| Skipping {absolute_name} since it is a symlink to a directory: {resolved_name}",
+                    err=True,
+                )
+            else:
+                # The following detection for unmerged codes comes from `man git-status`
+                if code == StatusCode.Unmerged:
+                    unmerged.append(resolved_name)
+                if (
+                    code[0] == StatusCode.Renamed
+                ):  # code is RXXX, where XXX is percent similarity
+                    removed.append(resolved_name)
+                    fname = status_output[2]
+                    trim_size += 1
+                    added.append(resolved_name)
+                if code == StatusCode.Added:
+                    added.append(resolved_name)
+                if code == StatusCode.Modified:
+                    modified.append(resolved_name)
+                if code == StatusCode.Deleted:
+                    removed.append(resolved_name)
```

Comment:
```suggestion
                continue

            # The following detection for unmerged codes comes from `man git-status`
            if code == StatusCode.Unmerged:
                unmerged.append(resolved_name)
            if (
                code[0] == StatusCode.Renamed
            ):  # code is RXXX, where XXX is percent similarity
                removed.append(resolved_name)
                fname = status_output[2]
                trim_size += 1
                added.append(resolved_name)
            if code == StatusCode.Added:
                added.append(resolved_name)
            if code == StatusCode.Modified:
                modified.append(resolved_name)
            if code == StatusCode.Deleted:
                removed.append(resolved_name)
```

just to get rid of a level of indentation

Reply:
Continue will actually cause an infinite loop cause the loop condition needs to be updated